Merge pull request #167 from jeffreywallphd/FALL2024-dev-combined(BA5…

Rambagaria opened the pull request:
…200/MIS4000)

Fall2024 dev combined(ba5200/mis4000)

## Using this Pull Request Template
1. In the Description of Changes section below, please fill out the details requested
2. Create the pull request
3. Check the boxes in the Type of Change section that apply to your changes
4. Check the boxes in the Checklist Before Review section that apply
5. Assign at least two reviewers to review your pull request by clicking the settings icon next to Reviewers

## Description of Changes
Please document the following about this pull request:
### Identify the organization for which you completed this work
If this work is associated with a university course, please identify the university/course/semester (ex. MTU/MIS3200/Fall2023).
* List the organizaiton here: 

### Identify what team you are on
* List the team here:

### Identify the task, story, or GitHub issue associated with this work
* Identify the task/story/issue here:

### Identify the particular changes that you made for this pull request
* Describe the changes here: 

## Type of Change (check all that apply)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation addition/update (non-breaking change that adds documentation to the system)

## Checklist Before Review
- [ ] Do you want to keep your branch for future changes or delete it once it is reviewed?
- [ ] I have followed the git practices outlined [here](GitPractices.md) to ensure code conflicts are minimized
- [ ] My code follows the style guidelines of this project outlined [here](documentation/StyleGuide.md)
- [ ] I have performed a self-review of my code. It runs without errors and follows the [Best Coding Practices](BestCodingPractices.md)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] There are no API keys, passwords, usernames, or other secrets in the code. I am using .env files for all secrets.
- [ ] I have added tests that prove my fix is effective or that my feature works

yaman-aljnadi committed:
Merge pull request #167 from jeffreywallphd/FALL2024-dev-combined(BA5200/MIS4000)

Fall2024 dev combined(ba5200/mis4000)

Rambagaria committed:
Added code for Bot

New folder by the name AutoProp_Codes is created. Everyone can push in their code here utself

jeffreywallphd reviewed (changes requested):
Thanks for pushing this request. Unfortunately, we can accept the code as it currently stands. Your code includes an API key in it, which a security vulnerability. Instead of adding the API key directly into the code, store it in a .env file that exists only on your local computer. The GitHub repository .gitignore file doesn't allow .env files to be uploaded. So if you store the API key in a .env file and then read in the file, you will be able to access the key in a secure manner.

Rambagaria committed:
API Key removal from main script

Stored the API key into local

Rambagaria commented:
As told, I have stored api key for Alphavantage locally in .env file and ignored the file in .gitignore

jeffreywallphd reviewed (changes requested):
Thank you for replacing the API key. There is still an authorization bearer token in the fetch_stock_name_ticker.py file. According to your documentation, this is the token for your personal account. You need to protect this. Also, is the Eden AI API access free to users? At the moment, we are only including free resources in this system to ensure that it meets the goals of the project. Please complete the full pull-request template as well. The template asks you to check for keys, passwords, and other secrets.

Daniel1999Akama committed:
voiceflow chatbot prototype 1

To access the prototype, download the file and upload on voiceflow.

Daniel1999Akama committed:
initial chatbot integration with html homepage

Daniel1999Akama committed:
API running locally with small HTML,CSS,JS UI

karthikkumarcholleti committed:
Add files via upload

This commit includes three fine-tuned BERT models, each designed to enhance query handling for stock-related questions:

1. **Intent Classification Model:** 
   - This model distinguishes between different types of stock-related intents such as `price inquiry`, `stock comparison`, `stock trend`, and `summary`. 
   - It allows the AI assistant to interpret the user's intent accurately, ensuring the correct response type is provided.

2. **Price Type Model:**
   - This model is fine-tuned to recognize various price-related terms in queries, such as `open`, `close`, `high`, `low`, `volume`, and `current`.
   - By identifying the price type, the assistant can extract and present specific stock price data requested by the user.

3. **Time Period Model:**
   - This model helps interpret time-related expressions like `last week`, `3 months ago`, `year to date`, and more.
   - It enables the assistant to narrow down data retrieval to the relevant time frame, improving the accuracy and relevance of responses.

**Technical Details:**
- These models were fine-tuned using the Hugging Face Transformers library.
- Training datasets include customized queries to ensure robustness across diverse phrasing and user intents.
- The models are saved in the `fine_tuned_models/` directory, with each model having its own subdirectory containing the `pytorch_model.bin` file and tokenizer configuration.

**Usage:**
To use these models, load them with the Transformers library:
```python
from transformers import BertForSequenceClassification, BertTokenizer

intent_model = BertForSequenceClassification.from_pretrained("path/to/intent_model")
intent_tokenizer = BertTokenizer.from_pretrained("path/to/intent_model")

price_model = BertForSequenceClassification.from_pretrained("path/to/price_model")
price_tokenizer = BertTokenizer.from_pretrained("path/to/price_model")

time_model = BertForSequenceClassification.from_pretrained("path/to/time_model")
time_tokenizer = BertTokenizer.from_pretrained("path/to/time_model")

Rambagaria committed:
Fetching stock Ticker

Eden API was having a freemium model, so working on fetching ticker symbols without using Eden API

Rambagaria committed:
Merge branch 'FALL2024-BA5200-Team5' of https://github.com/jeffreywallphd/AutoProphet into FALL2024-BA5200-Team5

karthikkumarcholleti committed:
updated from main

karthikkumarcholleti committed:
Added main code and fine tuned code with data set

Rambagaria committed:
Fetching tickers from CSV

The eden API Approach is discarded and a csv with around 300K companies are considered

Rambagaria committed:
Changed file name

Rambagaria committed:
nasdaq.csv is the dataset used

The previous file of dataset had duplications and very less uniqueness. So, had to find a new one. Also, the script for fetching ticker without using EDEN API is done
